From aae5220d961a419a1e160de90ee5c393c7c13607 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Mon, 6 Jun 2011 22:59:19 +0200
Subject: [PATCH 3/4] cryptsetup-generator: fix /etc/cryptsetup options

cryptsetup-generator parses the options in /etc/cryptsetup incorrectly.
It fails to find the 'swap' option in
  swap,foo
and instead it matches on
  swaplalala,foo

The condition for the comma separator is reversed.

https://bugzilla.redhat.com/show_bug.cgi?id=710839
---
 src/cryptsetup-generator.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/cryptsetup-generator.c b/src/cryptsetup-generator.c
index 696f44a..db8ebdf 100644
--- a/src/cryptsetup-generator.c
+++ b/src/cryptsetup-generator.c
@@ -47,7 +47,7 @@ static bool has_option(const char *haystack, const char *needle) {
                         continue;
                 }
 
-                if (f[l] != 0 && f[l] == ',') {
+                if (f[l] != 0 && f[l] != ',') {
                         f++;
                         continue;
                 }
-- 
1.7.4.4

